.gitignore

```diff
@@ -1,9 +1,18 @@
+# General ignore patterns
 *
 !*.*
 !*/
+
+# Ignore build artifacts and temporary files
 *.o
 *.so
 *.a
 *.asv
 *~
-doc*/
+
+# Ignore Sphinx build output but include source files
+doc/build/
+doc/api_docs/cpp
+
+# Add makefile 
+!Makefile
```

Comment:
Hi @pritkc , thank you for updating the `.gitignore` as suggested. 
What is the reason why we want to add the exclusion for the top-level `Makefile`? Can we just remove this exclusion line (i.e., always make sure that we track changes to the `Makefile`)? 

Also, please make sure to add IDE-specific automatically generated files. You PR now seems to be adding a few hidden files in `.DS_Store` (see screenshot) we want to avoid these.

![image](https://github.com/user-attachments/assets/f3b0a245-f876-4397-9e48-3ecda4587b1a)

```suggestion
# ignore vscode artifacts
*.vscode
*.DS
```